tests/plugins/test_koji_import.py

```diff
@@ -1300,43 +1300,105 @@ def test_koji_import_add_help(self, tmpdir, os_env, expect_result):
-        ["application/json", "application/vnd.docker.distribution.manifest.v1+json",
-         "application/vnd.docker.distribution.manifest.v2+json",
-         "application/vnd.docker.distribution.manifest.list.v2+json"],
-        # manifest lists, with compatibility formats back to V2schema1, without V1
-        ["application/vnd.docker.distribution.manifest.v1+json",
-         "application/vnd.docker.distribution.manifest.v2+json",
-         "application/vnd.docker.distribution.manifest.list.v2+json"],
+    @pytest.mark.parametrize(('config', 'expected'), [
+        ({'pulp_push': False,
+          'pulp_pull_in_worker': False,
+          'pulp_pull_in_orchestrator': False,
+          'schema1': False,
+          'schema2': False,
+          'list': False},
+         None),
+        ({'pulp_push': True,
+          'pulp_pull_in_worker': False,
+          'pulp_pull_in_orchestrator': False,
+          'schema1': False,
+          'schema2': False,
+          'list': False},
+         ["application/json"]),
+        ({'pulp_push': True,
+          'pulp_pull_in_worker': True,
+          'pulp_pull_in_orchestrator': False,
+          'schema1': True,
+          'schema2': False,
+          'list': False},
+         ["application/json",
+          "application/vnd.docker.distribution.manifest.v1+json"]),
+        ({'pulp_push': True,
+          'pulp_pull_in_worker': True,
+          'pulp_pull_in_orchestrator': False,
+          'schema1': True,
+          'schema2': True,
+          'list': False},
+         ["application/json",
+          "application/vnd.docker.distribution.manifest.v1+json",
+          "application/vnd.docker.distribution.manifest.v2+json"]),
+        ({'pulp_push': True,
+          'pulp_pull_in_worker': False,
+          'pulp_pull_in_orchestrator': True,
+          'schema1': True,
+          'schema2': True,
+          'list': False},
+         ["application/json",
+          "application/vnd.docker.distribution.manifest.v1+json",
+          "application/vnd.docker.distribution.manifest.v2+json"]),
+        ({'pulp_push': True,
+          'pulp_pull_in_worker': False,
+          'pulp_pull_in_orchestrator': True,
+          'schema1': True,
+          'schema2': True,
+          'list': True},
+         ["application/json",
+          "application/vnd.docker.distribution.manifest.v1+json",
+          "application/vnd.docker.distribution.manifest.v2+json",
+          "application/vnd.docker.distribution.manifest.list.v2+json"]),
+        ({'pulp_push': True,
+          'pulp_pull_in_worker': False,
+          'pulp_pull_in_orchestrator': True,
+          'schema1': True,
+          'schema2': True,
+          'list': True},
+         ["application/json",
+          "application/vnd.docker.distribution.manifest.v1+json",
+          "application/vnd.docker.distribution.manifest.v2+json",
+          "application/vnd.docker.distribution.manifest.list.v2+json"]),
```

Comment:
Duplicated test case?

Reply:
Reworked to test "schema 2 + manifest list" case